### ISSUE TITLE
EpubServer: Don't instantiate X handlers per epub being served

### ISSUE THREAD

**Issue by camill-a** [readium/r2-streamer-swift#3]
Currently, for each epub being served the EpubServer object creates 2 handlers, one for the manifest and one for the ressources .

As it is right now if it serves 20 epubs, there will be 40 handlers.

Instead, the server should only have 2 (or more, depending of the needs) handlers, and they should be called with a parameter (a hash or some id -  TBD) to decide which ressource they serve.
The routing to the ressource would be made inside the handler() instead of being directly made in the URL.

That would also fix the following problem: when wanting to remove an Epub being served from the server, we don't have the id associated to the handlers serving it, hence we can't remove these handlers.

**Comment by olivierkorner** [comment]
@camill-a Yes, indeed. 
The parameter should be a unique publication identifier, but I think it's the responsability of this API client to provide it (as it is now).

**Comment by HadrienGardeur** [comment]
Is this still relevant? cc @mickael-menu @aferditamuriqi

**Comment by mickael-menu** [comment]
Yes, it looks like it: https://github.com/readium/r2-streamer-swift/blob/630d8953cbad98fb229233caa9d98b2ce16474b3/r2-streamer-swift/Server/PublicationServer.swift#L140-L146

**Comment by mickael-menu** [comment]
Blocked until https://github.com/readium/r2-navigator-swift/issues/148 is resolved
